pubtools/_content_gateway/cgw_session.py

```diff
@@ -0,0 +1,96 @@
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+
+class CGWSession(object):
+    """Helper class to support cgw requests and authentication"""
+
+    def __init__(self, hostname, retries=3, verify=True, backoff_factor=2):
+        """
+        Args:
+            hostname (str)
+                hostname of CG service
+            retries (int)
+                number of http retries
+            verify (bool)
+                enable/disable SSL verification
+            backoff_factor (int)
+                backoff factor to apply between attempts after the second try
+        """
+        self.session = requests.Session()
+        self.hostname = hostname
+        self.verify = verify
+
+        retry = Retry(
+            total=retries,
+            read=retries,
+            connect=retries,
+            backoff_factor=backoff_factor,
+            status_forcelist=set(range(500, 512)),
+        )
+        adapter = HTTPAdapter(max_retries=retry)
+        self.session.mount("http://", adapter)
+        self.session.mount("https://", adapter)
+        self.session.headers["Content-type"] = 'application/json'
+        self.session.headers["Accept"] = 'application/json'
+
+    def get(self, endpoint, **kwargs):
+        """HTTP get request against CGW server API
+
+        Args:
+            endpoint (str)
+                API specific endpoint for the request
+        Returns:
+            requests.Response
+        """
+
+        return self.session.get(self._api_url(endpoint), verify=self.verify, **kwargs)
```

Comment:
I wolder what would happen if someone decided to specify verify in kwargs

Reply:
That we have to handle separately.